modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -379,7 +380,8 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
         total_stopw.stop("TotalTrain")
 
         self._info(f"Finished training: {self._num_samples} samples, {batch_number + 1} batches.")
-        self._log["num_samples"] = self._num_samples
+        self._log["num_samples_seen"] = self._num_samples
```

Comment:
I think it still somewhat is inconsistent now. 

When using no downsampling, num_samples_trained == num_samples_seen. 

When using BtS, num_samples_trained is lower than num_samples_seen, and we report num_samples_seen to the client.

When using StB, num_samples_trained == num_samples_seen again and we report num_samples_seen. This is why I proposed to just report the number of samples we'll train on to the client, because in StB we do that already anyways, unless I am missing something. I might also be overlooking something, but I think the three cases Stb/Bts/no sampling should be handled in the same way

Reply:
I didn't completely get your points:
When using StB, how do you define `num_samples_seen`? Should it be the same as `num_samples_trained`?

What do you think is inconsistent under the current changes?

Reply:
My point is that in BtS and StB, we train and see the same number of samples. As a user, I don't expect different numbers to show up in my client depending on whether I use BtS or StB. My intuition would be to show the number of training samples. If we show the number of samples seen, ok, but currently, in BtS, we show a progress bar with a higher absolute limit than in StB. But whatever, it's just visual. I guess we should focus on sth else

Reply:
Currently as we only report the number of samples seen, in BtS and StB the number is the same, so no unexpected inconsistency will appear right?

I agree with you that reporting only number of samples seen to the user is suboptimal, but anyway this PR is an improvement as it fixes an issue (also @robinholzi 's requirement is addressed), can we approve / merge this first? :D

wanna work on some other real problems first